pkg/types/validation/featuregate_test.go

```diff
@@ -0,0 +1,115 @@
+package validation
+
+import (
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+
+	v1 "github.com/openshift/api/config/v1"
+	"github.com/openshift/installer/pkg/types"
+	"github.com/openshift/installer/pkg/types/gcp"
+	"github.com/openshift/installer/pkg/types/vsphere"
+)
+
+func TestFeatureGates(t *testing.T) {
+	cases := []struct {
+		name          string
+		installConfig *types.InstallConfig
+		expected      string
+	}{
+		{
+			name: "GCP UserTags is allowed with Feature Gates enabled",
+			installConfig: func() *types.InstallConfig {
+				c := validInstallConfig()
+				c.FeatureSet = v1.TechPreviewNoUpgrade
+				c.GCP = validGCPPlatform()
+				c.GCP.UserTags = []gcp.UserTag{{ParentID: "a", Key: "b", Value: "c"}}
+				return c
+			}(),
+		},
+		{
+			name: "GCP UserTags is not allowed without Feature Gates",
+			installConfig: func() *types.InstallConfig {
+				c := validInstallConfig()
+				c.GCP = validGCPPlatform()
+				c.GCP.UserTags = []gcp.UserTag{{ParentID: "a", Key: "b", Value: "c"}}
+				return c
+			}(),
+			expected: `^platform.gcp.userTags: Forbidden: this field is protected by a feature gate which must be enabled through either the TechPreviewNoUpgrade or CustomNoUpgrade feature set$`,
+		},
```

Comment:
I'm ambivalent about adding these negative test cases, because they will need to be removed when the api is bumped to add the feature to the default feature set. I did add them in this case because I wanted to ensure the gating actually works.

Reply:
> Could you structure the code in such a way that you can use a fake set of features/featuresets in testing to test the logic without the moving base from o/api?

Yes I was considering the same thing and think this is a good alternative. My own personal inclination is to implement the individual test as it protects against other mistakes that could happen (users select the wrong field). IMHO I think this is good for now, but perhaps something we could refine if needed.